WooCommerce/UITestsFoundation/Screens/Orders/AddProductScreen.swift

```diff
@@ -21,4 +27,16 @@ public final class AddProductScreen: ScreenObject {
         app.buttons.staticTexts[name].tap()
         return try UnifiedOrderScreen()
     }
+
+    /// Selects multiple products from the list.
+    /// - Returns: Unified Order screen object.
+    @discardableResult
+    public func selectMultipleProducts(byName names: [String]) throws -> UnifiedOrderScreen {
+        app.buttons.staticTexts[names[0]].tap()
+        // TODO: Add further scenarios in next iterations: https://github.com/woocommerce/woocommerce-ios/issues/8997
+        if doneButton.exists {
+            doneButton.tap()
+        }
+        return try UnifiedOrderScreen()
+    }
```

Comment:
We should be able to get this method to work with the new screen already by doing something like this:
```
    public func selectMultipleProducts(numberOfProductsToAdd: Int) throws -> UnifiedOrderScreen {
        for number in 0...numberOfProductsToAdd-1 {
            // add accessibilityIdentifier for the product items on list and find buttons matching that identifier
            let products = app.buttons.matching(identifier: "product-item")
            products.element(boundBy: number).tap()
        }
        if doneButton.exists {
            doneButton.tap()
        }
        return try UnifiedOrderScreen()
    }
```

The parameter accepts the number of products the test should select instead of an array of product names (which adds a layer of complexity as that would depend on the values from the mock file, which could change), so from the test it would be something like:
```
...
.editOrderStatus()
.addProducts(numberOfProductsToAdd: 1)
.addCustomerDetails(name: order.billing.first_name)
...
```

The `accessibilityIdentifier` for `"product-item"` can be added after this line:
https://github.com/woocommerce/woocommerce-ios/blob/6b580f7ea8ec95a9278517091c965bffbd6e1c62/WooCommerce/Classes/ViewRelated/Products/ProductSelector/ProductSelectorView.swift#L74

This way, we don't have to revisit this method again when creating the new test to cover the multi-selection scenario. And since this change is pretty straightforward, I would recommend making it now rather than later.

Reply:
Thanks for the review! That sounds good, changed the method to accept a number of products rather than specific product names here: https://github.com/woocommerce/woocommerce-ios/pull/8989/commits/351e139a65cf8966bbfe387d8195f61ed96b3f77